e2e/test/prerequisites/E2ETestsSetup/e2eTestsSetup.ps1

```diff
@@ -0,0 +1,477 @@
+   az iot dps enrollment delete -g $ResourceGroup --dps-name $DeviceProvisioningServiceName --enrollment-id $IndividualEnrollmentId
+}
+Write-Host "`nAdding individual enrollment $IndividualEnrollmentId for device $individualDeviceId"
+az iot dps enrollment create -g $ResourceGroup --dps-name $DeviceProvisioningServiceName --enrollment-id $IndividualEnrollmentId --device-id $individualDeviceId --attestation-type x509 --certificate-path $IndividualDeviceCertPath --output none
+
+#################################################################################################################################################
+# Configure an AAD app and create self signed certs and get the bytes to generate more content info.
+#################################################################################################################################################
+
+$appId = az ad app list --show-mine --query "[?displayName=='$AppRegistrationName'].appId" --output tsv
+if($appId -eq $null)
+{
+    Write-Host "`nCreating App Registration $AppRegistrationName"
+    $appId = az ad app create --display-name $AppRegistrationName --reply-urls https://api.loganalytics.io/ --available-to-other-tenants false --query 'appId' --output tsv --output none
+    Write-Host "`nApplication $AppRegistrationName with Id $appId was created successfully."
+}
+$appId = az ad app list --show-mine --query "[?displayName=='$AppRegistrationName'].appId" --output tsv
+
+$spExists = az ad sp list --show-mine --query "[?appId=='$appId'].appId" --output tsv
+if($spExists -eq $null)
+{
+    Write-Host "`nCreating the service principal for the app registration if it does not exist"
+    az ad sp create --id $appId --output none
+}
+
+# The Service Principal takes a while to get propogated and if a different endpoint is hit before that, trying to grant a permission will fail.
+# Adding retries so that we can grant the permissions successfully without re-running the script.
+$retry = $true
+$retryCount = 1;
+while($retry -eq $true)
+{
+    try
+    {
+        Write-Host "`nGranting $appId Reader role assignment to the $Resourcegroup resource group - Trial $retryCount"
+        az role assignment create --role Reader --assignee $appId --resource-group $ResourceGroup --output none
+        $retry = $false
+    }
+    catch
+    {
+        $retryCount++
+        if($retryCount -eq 10)
+        {
+            Write-Host "Max retries reached"
+            $retry = $false
+        }
+    }
+}
+
+Write-Host "`nCreating a self signed certificate and placing it in $ResourceGroup"
+az ad app credential reset --id $appId --create-cert --keyvault $ResourceGroup --cert $ResourceGroup --output none
+Write-Host "`nSuccessfully created a self signed certificate for your application $AppRegistrationName in $ResourceGroup Keyvault: Cert name : $ResourceGroup";
+
+Write-Host "`nFetching the certificate binary"
+if(Test-Path ".\selfSignedCerts" -PathType Leaf){
```

Comment:
also space after the end paren before the curly brace

Reply:
I reformatted all files in VS Code. Somehow working in the PS editor does not translate to good code formatting.